src/mount/mount-tool.c

```diff
@@ -680,28 +696,194 @@ static int stop_mount(
         if (!arg_quiet)
                 log_info("Stopped unit %s%s%s for mount point: %s%s%s",
                          ansi_highlight(), mount_unit, ansi_normal(),
-                         ansi_highlight(), arg_mount_where, ansi_normal());
+                         ansi_highlight(), where, ansi_normal());
 
         return 0;
 }
 
 static int stop_mounts(
                 sd_bus *bus,
-                char **argv) {
+                char *where,
+                bool ignore_enoent) {
 
         int r;
 
-        r = stop_mount(bus, argv + optind, ".mount");
+        path_kill_slashes(where);
+
+        if (path_equal(where, "/")) {
+                log_error("Refusing to operate on root directory.");
+                return -EINVAL;
+        }
+
+        if (!path_is_safe(where)) {
+                log_error("Path contains unsafe components.");
```

Comment:
And this too, especially when multiple arguments are given it's important to give a hint to the user which one we don't like.